src/ns/adapters/outbound/lox24_client.py

```diff
@@ -0,0 +1,102 @@
+    lox24_port: int = Field(default=443, description="The port of the lox24 API")
+    lox24_token: SecretStr = Field(default=..., description="The authentication token")
+
+    lox24_timeout: PositiveFloat | None = Field(
+        default=10,
+        description=(
+            "The maximum amount of time (in seconds) to wait for a connection to the"
+            + " lox24 API. If set to `None`, the operation will wait indefinitely."
+        ),
+    )
+    lox24_sms_send_path: str = Field(
+        default="/sms", description="The path for sending SMS messages"
+    )
+    lox24_auth_token_header: str = Field(
+        default="X-LOX24-AUTH-TOKEN",
+        description="The header for the authentication token",
+    )
+    lox24_sender_id: str = Field(
+        default="GHGA", description="The sender ID to use when sending SMS messages"
+    )
+
+
+class Lox24Client(SmsClientPort):
+    """Concrete implementation of an SmsClientPort for the LOX24 SMS gateway."""
+
+    def __init__(self, *, config: Lox24ClientConfig):
+        """Assign config, which should contain all needed info"""
+        self._config = config
+        self._response: Response | None = None
+        self._send_sms_url: str = f"https://{self._config.lox24_host}:{self._config.lox24_port}/{self._config.lox24_sms_send_path.lstrip('/')}"
+        self._json_data: dict[str, str] = {"sender_id": self._config.lox24_sender_id}
+        self._headers: dict[str, str] = {
+            self._config.lox24_auth_token_header: self._config.lox24_token.get_secret_value()
+        }
+
+    def _raise_for_status(self, response: Response) -> None:
+        """Raise an exception if the response indicates an error."""
+        if response:
+            try:
+                response.raise_for_status()
+            except HTTPStatusError as e:
+                match e.response.status_code:
+                    case 400 | 404 | 422:
+                        raise SmsClientPort.RequestError() from e
+                    case 401 | 402 | 403 | 429:
+                        raise SmsClientPort.AccountError() from e
+                    case 500 | 502 | 503 | 504:
+                        raise SmsClientPort.SystemError() from e
+                    case _:
+                        raise SmsClientPort.GeneralSmsException(
+                            error_info=str(e)
+                        ) from e
+
+    def send_sms_message(self, message: dict):
+        """Send an SMS message to the Lox24 API."""
+        json_data = self._json_data | message
+        log.info(f"Sending SMS to {json_data['phone']}.")
+        response = post(
+            self._send_sms_url,
+            headers=self._headers,
+            json=json_data,
+            timeout=self._config.lox24_timeout,
+        )
+        if response.status_code != 201:
+            self._raise_for_status(response)
+        uuid = response.json().get("uuid", "unknown")
+        log.info(f"SMS sent to {json_data['phone']}. Response UUID {uuid}")
```

Comment:
Since this function expects `phone` and, realistically, `text` to be part of `message`, then I think `message` should be a more explicitly defined type (like the SmsNotification) or broken out into two parameters.
```suggestion
    def send_sms_message(self, *, phone: str, text: str):
        """Send an SMS message to the Lox24 API."""
        json_data = {
            "phone": phone,
            "text": text,
            **self._json_data,
        }
        log.info(f"Sending SMS to {phone}.")
        response = post(
            self._send_sms_url,
            headers=self._headers,
            json=json_data,
            timeout=self._config.lox24_timeout,
        )
        if response.status_code != 201:
            self._raise_for_status(response)
        uuid = response.json().get("uuid", "unknown")
        log.info(f"SMS sent to {phone}. Response UUID {uuid}")
```